internal/backend/local/test.go

```diff
@@ -345,930 +339,47 @@ func (runner *TestFileRunner) walkGraph(g *terraform.Graph, file *moduletest.Fil
 		sem.Acquire()
 		defer sem.Release()
 
-		switch v := v.(type) {
-		case *graph.NodeTestRun: // NodeTestRun is also executable, so it has to be first.
-			file := v.File()
-			run := v.Run()
-			if file.GetStatus() == moduletest.Error {
-				// If the overall test file has errored, we don't keep trying to
-				// execute tests. Instead, we mark all remaining run blocks as
-				// skipped, print the status, and move on.
-				run.Status = moduletest.Skip
-				runner.Suite.View.Run(run, file, moduletest.Complete, 0)
-				return
-			}
-			if runner.Suite.Stopped {
-				// Then the test was requested to be stopped, so we just mark each
-				// following test as skipped, print the status, and move on.
-				run.Status = moduletest.Skip
-				runner.Suite.View.Run(run, file, moduletest.Complete, 0)
-				return
-			}
-
-			// TODO: The execution of a NodeTestRun is currently split between
-			// its Execute method and the continuation of the walk callback.
-			// Eventually, we should move all the logic related to a test run into
-			// its Execute method, effectively ensuring that the Execute method is
-			// enough to execute a test run in the graph.
-			diags = v.Execute(runner.EvalContext)
-			if diags.HasErrors() {
-				return diags
-			}
-
-			startTime := time.Now().UTC()
-			runner.run(run, file, startTime)
-			runner.Suite.View.Run(run, file, moduletest.Complete, 0)
-			file.UpdateStatus(run.Status)
-		case graph.GraphNodeExecutable:
-			diags = v.Execute(runner.EvalContext)
-			return diags
-		default:
-			// If the vertex isn't a test run or executable, we'll just skip it.
-			return
+		// Bind the terraform context options to the node
+		if v, ok := v.(graph.BindContextOpts); ok {
+			v.BindContextOpts(runner.Suite.Opts)
```

Comment:
Given you have the suite opts available when the graph is built, can you provide these as input into the transformers at that point instead of dynamically loading them here?